siriuspy/siriuspy/clientarch/client.py

```diff
@@ -286,10 +292,19 @@ def getPVDetails(self, pvname, get_request_url=False):
         resp = self._make_request(url, return_json=True)
         return None if not resp else resp
 
+    def switch_to_online_data(self):
+        """."""
+        self.server_url = _envars.SRVURL_ARCHIVER
+
+    def switch_to_offline_data(self):
+        """."""
+        self.server_url = _envars.SRVURL_ARCHIVER_OFFLINE_DATA
+
```

Comment:
If you logged in and then switched the data source, the `session` in use will be the wrong one.

I think the `ClientArchiver.switch_to_*_data` methods should also set `self.session = None`. That way, `_handle_request` will automatically let you know you missed a `login` call, if necessary, after calling some of the `switch_to_*_data` methods.

```suggestion
    def switch_to_online_data(self):
        """."""
        self.server_url = _envars.SRVURL_ARCHIVER
        self.session = None

    def switch_to_offline_data(self):
        """."""
        self.server_url = _envars.SRVURL_ARCHIVER_OFFLINE_DATA
        self.session = None

```

Reply:
you are right, @ericonr . thanks.

Reply:
this 'suggested change' ready for commit is very nice! how do I use it ?

Reply:
@xresende when reviewing the code, you can select a range and sometimes (sadly, not always) an "add suggestion" button shows up in the top bar, alongside the formatting buttons. I don't know if you need to be reviewing by commits and not by all changes for that to happen.

![image](https://github.com/lnls-sirius/dev-packages/assets/34201958/ceac68c9-a0b8-4e0e-8a9e-a8faeb23bdec)

Reply:
thanks.